src/Service/PaymentMethodsService.php

```diff
@@ -104,18 +114,29 @@ public function __construct(
      */
     public function getPaymentMethods(SalesChannelContext $context, $orderId = ''): array
     {
+        $requestData = $this->buildPaymentMethodsRequestData($context, $orderId);
+
+        $paymentRequestString = json_encode($requestData);
+        $cacheKey = 'adyen_payment_methods_' . md5($paymentRequestString);
```

Comment:
Better to use [uniqid](https://www.php.net/manual/en/function.uniqid.php) here, as md5 doesn't guarantee uniqueness

Reply:
discussed. using md5 so that it would match if the requestData array is the same